projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/components/linked-cases-table/linked-cases-to-table.component.ts

```diff
@@ -0,0 +1,164 @@
+    this.getAllLinkedCaseInformation();
+    if (this.route.snapshot.data.case) {
+      this.linkedCasesService.caseDetails = this.route.snapshot.data.case;
+    }
+  }
+
+  public getCaseRefereneLink(caseRef) {
+    return caseRef.slice(this.caseId.length - 4);
+  }
+
+  public sortLinkedCasesByReasonCode(searchCasesResponse) {
+    const topLevelresultArray = [];
+    let secondLevelresultArray = [];
+    searchCasesResponse.forEach((item: any) => {
+      const reasons = item && item.reasons || [];
+      const consolidatedStateReason = reasons.map(x => x).find(
+        reason => reason === LinkedCasesToTableComponent.CASE_CONSOLIDATED_REASON_CODE
+      );
+      const progressedStateReason = reasons.map(x => x).find(
+        reason => reason === LinkedCasesToTableComponent.CASE_PROGRESS_REASON_CODE
+      );
+      let arrayItem;
+      if (progressedStateReason) {
+        arrayItem = { ...item };
+        topLevelresultArray.push(arrayItem);
+      } else if (consolidatedStateReason) {
+        arrayItem = { ...item };
+        secondLevelresultArray = [{ ...item }, ...secondLevelresultArray]
+      } else {
+        arrayItem = { ...item };
+        secondLevelresultArray.push({ ...item });
+      }
+    })
+    return topLevelresultArray.concat(secondLevelresultArray)
+  }
+
+  public getAllLinkedCaseInformation() {
+    const searchCasesResponse = [];
+    const caseFieldValue = this.caseField && this.caseField.id === 'caseLinks' ? this.caseField.value : [];
+    // Generate the list of observables
+    caseFieldValue.forEach(fieldValue => {
+      if (fieldValue && fieldValue.id) {
+        searchCasesResponse.push(this.casesService.getCaseViewV2(fieldValue.id));
+      }
+    });
+    if (searchCasesResponse.length) {
+      this.searchCasesByCaseIds(searchCasesResponse).subscribe((searchCases: any) => {
+        const casesResponse = [];
+        searchCases.forEach(response => {
+          casesResponse.push(this.mapResponse(response))
+        });
+        this.linkedCasesFromResponse = this.sortLinkedCasesByReasonCode(casesResponse);
+        this.isLoaded = true;
+        const caseLinks = this.linkedCasesFromResponse.map(item => {
+          return {
+            caseReference: item.caseReference,
+            caseName: item.caseName,
+            caseService: item.service,
+            caseType: item.caseType,
+            unlink: false,
+            reasons: item.reasons && item.reasons.map(reason => {
+              return {
+                reasonCode: reason
+              } as LinkReason
+            }),
+          } as CaseLink
+        });
+        this.linkedCasesService.initialCaseLinks = caseLinks;
+        this.linkedCasesService.linkedCases = caseLinks;
+        this.isServerError = false;
+      },
+        err => {
+          this.isServerError = true;
+          this.notifyAPIFailure.emit(true)
+        }
+      );
+    }
+  }
+
+  public searchCasesByCaseIds(searchCasesResponse: any[]) {
```

Comment:
@johnbenjamin-hmcts - please specify return type if possible

Reply:
@Daniel-Lam I have updated the code now.